pandas/tests/indexes/base_class/test_pickle.py

```diff
@@ -0,0 +1,11 @@
+from pandas import Index
+import pandas._testing as tm
+
+
+def test_pickle_preserves_object_dtype():
```

Comment:
can you paramterize on all the index types? (i think we have a fixture)

Reply:
we can come back and add more tests that use the fixture, but id like to keep this test targeted at a specific bug